terraform/java/ec2/asg/main.tf

```diff
@@ -215,6 +216,19 @@ resource "null_resource" "remote_service_setup" {
       # The application needs time to come up and reach a steady state, this should not take longer than 30 seconds
       sleep 30
 
+      # Check if the application is up. If it is not up, then exit 1.
+      attempt_counter=0
+      max_attempts=30
+      until $(curl --output /dev/null --silent --head --fail $(echo "http://localhost:8080/healthcheck" | tr -d '"')); do
```

Comment:
Almost forgot, can you enforce a --max-time or something of the sort on this? We previously saw this retry take a total of 1h15 randomly instead of the expected 30*10. Put an appropriate limit using one or more of the time-related options for curl

Reply:
Done